get_soccer_players_articles.py

```diff
@@ -9,26 +9,44 @@
 
 DEBUG = True
 
-
+''' Returns a list with wiki ids from the input file filein'''
 def load_wiki_ids(filein):
     with open(filein) as i:
         return [l.strip() for l in i.readlines()]
 
-
+'''Prints out article present in path corpus directory if its ID is present in "soccer_ids" list'''
 def extract_soccer_articles(soccer_ids, corpus_dir, output_dir):
+    # Iterating over the files and sub-directories from corpus directory using os.walk function
     for path, subdirs, files in os.walk(corpus_dir):
+        # Iterating over all the files
         for name in files:
             f = os.path.join(path, name)
-            with open(f) as i:
+            # Opening a single file f
+            with open(f,'r') as i:
                 content = ''.join(i.readlines())
-            match = re.search('id="([^"]+)"', content)
-            current_id = match.group(1)
-            if DEBUG:
-                print "File = [%s] - Wiki ID = [%s]" % (f, current_id)
-            if current_id in soccer_ids:
-                shutil.copy(f, output_dir)
+            # Finding ALL the doc ids in content using a simple regex.
```

Comment:
The `get_soccer_players.py` script takes as input the corpus dir 
**after** the `csplit` run
So, **each** file corresponds to a **single** article.
You probably ran that script with a **non-split** corpus as input.

On 3/17/15 2:00 PM, Enayat Ullah wrote:
> In get_soccer_players_articles.py
> <https://github.com/dbpedia/fact-extractor/pull/16#discussion_r26569678>:
>
>>                  content = ''.join(i.readlines())
>> -            match = re.search('id="([^"]+)"', content)
>> -            current_id = match.group(1)
>> -            if DEBUG:
>> -                print "File = [%s] - Wiki ID = [%s]" % (f, current_id)
>> -            if current_id in soccer_ids:
>> -                shutil.copy(f, output_dir)
>> +            # Finding ALL the doc ids in content using a simple regex.
>
> @marfox <https://github.com/marfox> Yes, I did see the csplit function.
> But there is no explicit check for soocer_id's there. So a wiki doc in
> which out of ,say n articles, only 1 is related to soccer, is copied by
> the get_soocer_players_articles script. And the csplit in extract.sh
> just segregates it into multiple files. And so the articles totally
> unrelated to soccer end up in the final docs.( This is the result I
> obtained when I did run the code).
>
> This is what I recollect from the two scripts. I perhaps may be
> fundamentally wrong regarding it. Please correct me then. Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dbpedia/fact-extractor/pull/16/files#r26569678>.
>